File: php/state/Context.php
<?php

/**
 * 房间对象
 * 这里的上下文对象 可以理解成房间
 */

class Context
{
    private $state;

    public function setState($state)
    {
        echo '修改状态', PHP_EOL;

        $this->state = $state;

        $this->state->handle();
    }
}

File: java/pattern/src/state/Context.java
package state;

/**
 * 房间对象
 * 这里的上下文对象 可以理解成房间
 */
public class Context {
    private State state;

    public void setState(State state) {
        System.out.println("修改状态");
        this.state = state;

        this.state.handle();
    }
}
